### PR TITLE
Use delivery channel to get kafka producer metrics

### PR DESCRIPTION
**Pull request by agbpatro**
# Description
Confluent provides the channel interface to get delivery reports. We should use that to get kafka metrics

## Type of change

Please select all options that apply to this change:

- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [X] My code follows the style of this project.
- [X] I have performed a self-review of my code.
- [ ] I have made corresponding updates to the documentation.
- [ ] I have added/updated unit tests to cover my changes.
- [ ] I have added/updated integration tests to cover my changes.

### CODE REVIEW

File: datastore/kafka/kafka.go [hunk 1]
```diff
@@ -58,9 +60,11 @@ func NewProducer(config *kafka.ConfigMap, namespace string, reliableAckWorkers i
 		logger:            logger,
 		airbrakeHandler:   airbrakeHandler,
 		reliableAck:       reliableAckWorkers > 0,
+		deliveryChan:      make(chan kafka.Event),
+		ackChan:           ackChan,
```
Comment (aaronpkahn) [inline datastore/kafka/kafka.go:64]: I don't understand this design for ackChan.  Wouldn't we need an ackChan per connection?

Comment (agbpatro) [reply datastore/kafka/kafka.go:64]: you are right. I have disabled setting reliable ack config for now. Will remove the restriction when its supported properly

Comment (aaronpkahn) [reply datastore/kafka/kafka.go:64]: ok, well we should complete the implementation or remove ackChan. Let's not leave code/variables around that aren't being used

**Review by aaronpkahn — APPROVED** [review]

File: config/config_initializer.go [hunk 2]
```diff
@@ -54,6 +55,10 @@ func loadApplicationConfig(configFilePath string) (*Config, error) {
 	}
 	config.MetricCollector = metrics.NewCollector(config.Monitoring, logger)
 
+	// TODO disble this check when reliable acks are properly supported
```
Comment (ThomasAlxDmy) [inline config/config_initializer.go:58]: When is this going to be?

Comment (agbpatro) [reply config/config_initializer.go:58]: https://github.com/teslamotors/fleet-telemetry/pull/167

**Pull request merged**